byteparse/shared/src/test/scala/byteparse/BmpTests.scala

```diff
@@ -67,31 +68,30 @@ object BmpTests extends TestSuite {
           /*pixels*/  "FF 00 00 7F  00 FF 00 7F  00 00 FF 7F  00 00 FF 7F " +
                       "FF 00 00 FF  00 FF 00 FF  00 00 FF FF  FF FF FF FF")
 
-        val Parsed.Success(bmp2, _) = bmp.parse(file1)
 
-        assert(
-          compareBmps(
-            bmp2,
-            Bmp(
-              FileHeader(19778, 154, 122),
-              BitmapInfoHeader(BitmapInfoHeaderPart(4, 2, 1, 32, 3, 32, 2835, 2835, 0, 0)),
-              ArrayBuffer(
-                ArrayBuffer(
-                  Pixel(BS(0xff, 0, 0, 0xff)),
-                  Pixel(BS(0, 0xff, 0, 0xff)),
-                  Pixel(BS(0, 0, 0xff, 0xff)),
-                  Pixel(BS(0xff, 0xff, 0xff, 0xff))
-                ),
-                ArrayBuffer(
-                  Pixel(BS(0xff, 0, 0, 0x7f)),
-                  Pixel(BS(0, 0xff, 0, 0x7f)),
-                  Pixel(BS(0, 0, 0xff, 0x7f)),
-                  Pixel(BS(0, 0, 0xff, 0x7f))
-                )
-              )
+        val expected = Bmp(FileHeader(19778, 154, 122),
+          BitmapInfoHeader(BitmapInfoHeaderPart(4, 2, 1, 32, 3, 32, 2835, 2835, 0, 0)),
+          ArrayBuffer(ArrayBuffer(
+            Pixel(BS(0xff, 0, 0, 0xff)),
+            Pixel(BS(0, 0xff, 0, 0xff)),
+            Pixel(BS(0, 0, 0xff, 0xff)),
+            Pixel(BS(0xff, 0xff, 0xff, 0xff))),
+            ArrayBuffer(
+              Pixel(BS(0xff, 0, 0, 0x7f)),
+              Pixel(BS(0, 0xff, 0, 0x7f)),
+              Pixel(BS(0, 0, 0xff, 0x7f)),
+              Pixel(BS(0, 0, 0xff, 0x7f))
             )
           )
         )
+
+        val Parsed.Success(bmp2, _) = bmp.parse(file1)
+        assert(compareBmps(bmp2, expected))
+
+        for(chunkSize <- Seq(1, 4, 16, 64, 256, 1024)){
```

Comment:
Iterator parsing is approximately 2x slower than non-Iterator, so adding 6 more Iterator tests may significantly increase total time of all of these tests.